lib/src/shortest_path.dart

```diff
@@ -24,12 +24,12 @@ import 'dart:collection';
 ///
 /// If you supply one of [equals] or [hashCode], you should generally also to
 /// supply the other.
-List<T> shortestPath<T>(
+List<T>? shortestPath<T>(
```

Comment:
Should this be `<T extends Object>`? I don't know that `null` is ever useful value here... but maybe we don't want to be overly opinionated about it. There is an assert below that `start` is not null though.

Reply:
My gut reaction is that most of these should be `<T extends Object>`, but I could also be convinced otherwise.